src/packet_analysis/protocol/tcp/TCP.cc

```diff
@@ -130,7 +130,8 @@ void TCPAnalyzer::DeliverPacket(Connection* c, double t, bool is_orig, int remai
 	// Call DeliverPacket on the adapter directly here. Normally we'd call ForwardPacket
 	// but this adapter does some other things in its DeliverPacket with the packet children
 	// analyzers.
-	adapter->DeliverPacket(len, data, is_orig, adapter->LastRelDataSeq(), ip.get(), remaining);
+	adapter->DeliverPacket(remaining, data, is_orig, adapter->LastRelDataSeq(), ip.get(),
+	                       pkt->cap_len);
```

Comment:
Mainly because passing `remaining` twice seemed awkward, the doc string from `NextPacket()` spells out it's the capture length of the packet, it not making a difference as you found and passing `pkt->cap_len` minimally does provide additional information to downstream analyzers.

```
> 	 * @param len The number of bytes passed in.
...
> 	 * @param caplen The packet's capture length, if available.
```

I'm went forth and back now for a  revert it, but I think sticking with `pkt->cap_len` rather than passing remaining twice seems reasonable.

Reply:
Sounds good, thank you! Was just trying to understand the motivation.